packages/core-contracts/src/contracts/arks/SiUSDArk.sol

```diff
@@ -0,0 +1,190 @@
+     *      iUSD has 18 decimals, USDC has 6 decimals
+     * @return assets The total balance of USDC-equivalent assets
+     */
+    function totalAssets() public view override returns (uint256 assets) {
+        uint256 siUSDShares = siUSD.balanceOf(address(this));
+        if (siUSDShares > 0) {
+            // Convert siUSD shares to iUSD amount (18 decimals)
+            uint256 iUSDAmount = siUSD.convertToAssets(siUSDShares);
+            // Convert iUSD (18 decimals) to USDC equivalent (6 decimals)
+            // iUSD is pegged 1:1 to USD, so we just adjust decimals
+            assets = iUSDAmount / 1e12;
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Internal function to get the total assets that are withdrawable
+     * @dev Returns the USDC-equivalent value of assets that can be withdrawn
+     * @return withdrawableAssets Amount of USDC that can be withdrawn
+     */
+    function _withdrawableTotalAssets()
+        internal
+        view
+        override
+        returns (uint256 withdrawableAssets)
+    {
+        uint256 siUSDShares = siUSD.balanceOf(address(this));
+        if (siUSDShares > 0) {
+            // Get maximum iUSD we can withdraw from siUSD vault (18 decimals)
+            uint256 maxIUSD = siUSD.maxWithdraw(address(this));
+            // Convert to USDC equivalent (6 decimals)
+            withdrawableAssets = maxIUSD / 1e12;
+        }
+    }
+
+    /**
+     * @notice Deposits USDC into InfiniFi via Gateway
+     * @dev Uses Gateway.mintAndStake() to convert USDC -> siUSD directly
+     * @param amount The amount of USDC to deposit
+     */
+    function _board(uint256 amount, bytes calldata) internal override {
+        // Use Gateway to mint iUSD and stake to siUSD in one transaction
+        // This returns the amount of iUSD that was minted (and staked)
+        gateway.mintAndStake(address(this), amount);
+    }
+
+    /**
+     * @notice Withdraws USDC from siUSD vault via Gateway
+     * @dev Flow: siUSD -> iUSD (unstake) -> USDC (redeem)
+     * @param amount The amount of USDC to withdraw
+     */
+    function _disembark(uint256 amount, bytes calldata) internal override {
+        // Step 1: Calculate how much iUSD we need (convert USDC decimals to iUSD decimals)
+        uint256 iUSDNeeded = amount * 1e12; // USDC (6 dec) -> iUSD (18 dec)
+
+        // Step 2: Calculate how many siUSD shares we need to unstake to get that much iUSD
+        uint256 siUSDSharesToUnstake = siUSD.previewWithdraw(iUSDNeeded);
+
+        // Step 3: Unstake siUSD to receive iUSD via Gateway
+        uint256 iUSDReceived = gateway.unstake(
+            address(this),
+            siUSDSharesToUnstake
+        );
+
+        // Step 4: Redeem iUSD for USDC via Gateway (with 0 slippage protection for now)
```

Comment:
It says without slippage protection (for now). What does `for now` mean? Are we going to implement it? When?